### ISSUE TITLE
Απορίες για εκπαίδευση lstm - Ερώτημα 14

### ISSUE THREAD

**Issue by iliaspap** [slp-ntua/patrec-labs#130]
Καλησπέρα,
Έχουμε δυσκολευτεί λίγο στην εκπαίδευση του lstm στο βήμα 14, καθώς φαίνεται ότι το δίκτυο δε μαθαίνει, παρόλο που δοκιμάσαμε πολλά σύνολα παραμέτρων. Προβληματιζόμαστε σχετικά με τα εξής:
1) Όταν δοκιμάσαμε να κανονικοποιήσουμε τα χαρακτηριστικά, μειώθηκε πολύ το training loss, κάτι που ερμηνεύσαμε σαν σωστή εκπαίδευση του δικτύου. Ωστόσο οι προβλέψεις του στο test set ήταν πρακτικά τυχαίες. Γιατί συμβαίνει αυτό; Πώς καταλαβαίνουμε πότε κάνουμε κανονικοποίηση;
2) Νομίζω δεν έχουμε καταλάβει καλά την τάξη μεγέθους των παραμέτρων. Εμείς στην αρχή χρησιμοποιούσαμε παραμέτρους στην τάξη μεγέθους αυτών του προβλήματος στο βήμα 8 για την πρόβλεψη ακολουθιών συνημιτόνου (~20 hidden size, ~30 EPOCHS, num_layers = 1). Ωστόσο επειδή δε μαθαίνει το δίκτυο σκεφτόμαστε μήπως πρέπει να αυξήσουμε κατά πολύ αυτές τις παραμέτρους. Είναι πολύ μικρές αυτές οι τιμές; Ιδανικά θα κάναμε κάποιο grid search για να βρούμε τις καλύτερες, όμως ακόμη και γι'αυτό δε πρέπει να ξέρουμε την τάξη μεγέθους των παραμέτρων, αλλιώς η εκπαίδευση θα πάρει 1 μέρα... Πώς μπορούμε να γνωρίζουμε την τάξη μεγέθους των παραμέτρων του δικτύου για κάποιο πρόβλημα, και πόση περίπου είναι η αναμενόμενη διάρκεια μιας φυσιολογικής εκπαίδευσης;
3) Εκτός από το lstm layer και το classification layer χρειαζόμαστε κι άλλα layers;
Ευχαριστώ πολύ!
